File: src/main/scala/Task3.hs
-- Two Lists symmetric difference
main = do
    let s1 = [1..100]
    let s2 = [1..102]
    let d = diff s1 s2
    putStrLn $ show d

diff :: (Num a, Eq a) => [a] -> [a] -> [a]
diff f1 f2
  | length f1 > length f2 = foldl (\acc y -> if (y `elem` f2) then acc else acc ++ [y]) [] f1
  | otherwise = foldl (\acc y -> if (y `elem` f1) then acc else acc ++ [y]) [] f2

File: src/main/scala/Task3.scala
// Symmetric difference of two Seqs or Sets
object Task3 {
  def main(args: Array[String]): Unit = {
    println(task3(List("A", "C", "K", "G", "L", "M"), List("B", "M", "G", "V")))
    println(task3((1 to 100).toSet, (1 to 101).toSet))
  }

  def task3[T](s1: Seq[T], s2: Seq[T]): Seq[T] = {
    val seed: Seq[T] = Seq()
    s1.foldLeft(seed)((acc, x) => if (s2 contains x ) acc else acc :+ x)
  }

  def task3[T](s1: Set[T], s2: Set[T]): Set[T] = {
    val seed: Set[T] = Set()
    if (s1.size > s2.size)
      s1.foldLeft(seed)((acc, x) => if (s2 contains x) acc else acc + x)
    else
      s2.foldLeft(seed)((acc, x) => if (s1 contains x) acc else acc + x)
  }
}
